File: app/Season42/Library/Search/MovieDetails.swift
import Foundation

/// Everything the app shows about one movie the user opened from a search, as the BFF answers
/// with it: what it is called, what it is called where it was made, and what it is about. This
/// is the whole of it — the runtime and the ratings TMDB also knows are not part of the ask.
///
/// There are no seasons, which is the whole of what separates it from a Series Details: a movie
/// is one thing to watch, so nothing here is flattened and nothing about a copy is invented.
///
/// The BFF also answers whether there is a poster to be had, and nothing here reads it: a movie
/// carries no Poster yet, and a field the app has no use for is a field it does not decode. It
/// is the ask the Poster will be made with when there is one to make.
///
/// Someone else's data, never stored. Not even the id it was asked for comes back: the app
/// already has that from the Movie Match it opened, and nothing in the Library ever holds one
/// (ADR-0002).
struct MovieDetails: Decodable, Equatable, Sendable {
    let title: String

    /// What the movie is called where it was made, which for one made elsewhere is the thing
    /// that tells two similar English titles apart. The same as `title` often enough, and empty
    /// where TMDB carries none.
    let originalTitle: String

    /// What the movie is about, in TMDB's words. Empty where TMDB carries none.
    let overview: String
}

File: bff/Season42.Bff/MovieDetails.cs
namespace Season42.Bff;

/// <summary>
/// Everything the app shows about one movie it opened: what it is called, what it is called
/// where it was made, what it is about, and whether there is a poster to ask for. This is the
/// whole of it — the runtime, the budget and the release date TMDB also knows are not here,
/// because nothing on the detail screen shows them.
/// </summary>
/// <remarks>
/// There are no seasons, which is the whole of what separates this from a
/// <see cref="SeriesDetails"/>: a movie is one thing to watch, so nothing here is flattened and
/// nothing about it is invented.
/// </remarks>
/// <remarks>
/// TMDB's poster path is not among them. Whether there is a poster is a yes or a no, and the
/// poster itself is asked for by the id the app already has: a path the app could send is a path
/// this server would have to defend against, and there is none (ADR-0012). The yes or no is here
/// so the app can draw its placeholder without firing an ask it expects to be refused.
/// </remarks>
/// <remarks>
/// The id it was asked for is not answered back. The app already has it, from the
/// <see cref="MovieMatch"/> it opened, and it is never stored on either side: the Library holds
/// what the user copied, not a link back to someone else's record (ADR-0002).
/// </remarks>
public sealed record MovieDetails(
    string Title,
    string OriginalTitle,
    string Overview,
    bool HasPoster);
